### PR TITLE
Adding 'setup-python' and 'download-artifacts' options

### PR DESCRIPTION
**Pull request by germa89**
As the title.

Now if you want to use artifacts different than the defaults, you need to download them by yourself and locate them in the right folder. Then execute the appropriate action.

### CODE REVIEW

**Comment by germa89** [comment]
Some actions has ``skip-install`` which depending on the action can do just the ``pip install`` (i.e. ``doc-build`` action) or also the checkout, python install and project install (i.e. ``check-license`` action).

We should probably be consistent with its usage.

**Comment by germa89** [comment]
The ``release-to-PyPI`` and ``release-to-Github`` are not consistent.

The PyPI action download the artifacts (all) to a folder called ``${{ inputs.library-name}}-artifacts`` but then, the ``release-to-github`` action check the folder ``other-artifacts``.

**Comment by RobPasMue** [comment]
This is in the lines of #140.. we should figure out the strategy to follow first before merging anything. Pinging @jorgepiloto as well for acknowledgement.

**Comment by germa89** [comment]
> This is in the lines of #140.. we should figure out the strategy to follow first before merging anything. Pinging @jorgepiloto as well for acknowledgement.

I think this PR is fine for now. We can work out the path structure later.

File: doc-build/action.yml [hunk 1]
```diff
@@ -38,6 +38,7 @@ inputs:
       required: false
       type: boolean
 
+
```
Comment (RobPasMue) [inline doc-build/action.yml:41]:
```suggestion
```

File: code-style/action.yml [hunk 2]
```diff
@@ -18,6 +18,7 @@ inputs:
       required: false
       type: boolean
 
+
```
Comment (RobPasMue) [inline code-style/action.yml:21]:
```suggestion
```

File: doc-deploy-stable/action.yml [hunk 3]
```diff
@@ -86,8 +91,8 @@ runs:
         echo "VERSION=$major.$minor" >> $GITHUB_ENV
 
     - name: "Download the HTML documentation artifact in release/${{ github.ref_name }}"
-      if: env.ACCEPTED_FORMAT == 'true'
-      uses: actions/download-artifact@v3
+      if: ${{ env.ACCEPTED_FORMAT == 'true' && inputs.download-artifacts }}
```
Comment (RobPasMue) [inline doc-deploy-stable/action.yml:94]: Not sure if this will work. Pinging @jorgepiloto

File: _release-pypi/action.yml [hunk 4]
```diff
@@ -22,6 +22,12 @@ inputs:
     description: 'Desired Python version for using twine'
     required: true
     type: string
+  download-artifacts:
+    description: 'Whether or not download the artifacts. By default download an artifact named "``library-name``-artifacts"'
+    required: false
```
Comment (RobPasMue) [inline _release-pypi/action.yml:27]:
```suggestion
    required: true
```

File: _release-pypi/action.yml [hunk 5]
```diff
@@ -22,6 +22,12 @@ inputs:
...
+    type: bool
+    default: true
```
Comment (RobPasMue) [inline _release-pypi/action.yml:29]:
```suggestion
```

File: _release-pypi/action.yml [hunk 6]
```diff
@@ -22,6 +22,12 @@ inputs:
     description: 'Desired Python version for using twine'
     required: true
     type: string
+  download-artifacts:
```
Comment (RobPasMue) [inline _release-pypi/action.yml:25]: For "private" actions (i.e. starting with "_"), it is better not to have optional args.

**Review by RobPasMue — COMMENTED** [review]
I'm fine with it (for now)... but only because it is needed for your release @germa89.

Honestly, I am looking forward to #141 ... When the time comes, this might be reverted. Just letting you know.

I'll leave it for @jorgepiloto to take the final call.

File: _release-pypi/action.yml (hunk 6 above)
Comment (germa89) [reply _release-pypi/action.yml:25]: Fair enough

File: doc-deploy-stable/action.yml (hunk 3 above)
Comment (germa89) [reply doc-deploy-stable/action.yml:94]: Why? I think it should (if you mean the ``if`` syntax)

https://github.com/orgs/community/discussions/25641#discussioncomment-3248571

Comment (germa89) [reply doc-deploy-stable/action.yml:94]: You are right...

It should be without the ``${{ }}``.

**Comment by germa89** [comment]
Where are we on this??

**Comment by jorgepiloto** [comment]
Discarding these changes based on the strict philosophy of the actions.

**Pull request closed without merging**